src/luacheck/stages/resolve_locals.lua

```diff
@@ -79,18 +79,26 @@ local function in_scope(var, index)
 end
 
 local function contains_call(node)
+   if node.is_contains_call ~= nil then
```

Comment:
```suggestion
   if node._contains_call ~= nil then
```

I think this looks good my only concern is the naming of the cached key. The `is_` feels wrong to me because the node itself "is not" a "contains call", it is what it is and we're just caching the extra property. How about just using a private key prefix instead?

Reply:
hi, I just amended and force pushed this suggestion 👍 @alerque